src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,122 @@
+import { getPeople } from '../api';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { Loader } from './Loader/Loader';
+import cn from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+
+export const People = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoaging] = useState(false);
+  const [erorMesage, setErorMesage] = useState('');
+
+  const { slug } = useParams();
+
+  useEffect(() => {
+    setLoaging(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setErorMesage('Something went wrong');
+      })
+      .finally(() => setLoaging(false));
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {loading && <Loader />}
+
+          {erorMesage && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              {erorMesage}
+            </p>
+          )}
+
+          {!people.length && !loading && !erorMesage && (
+            <p data-cy="noPeopleMessage">There are no people on the server</p>
+          )}
+
+          {!!people.length && !loading && (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Sex</th>
+                  <th>Born</th>
+                  <th>Died</th>
+                  <th>Mother</th>
+                  <th>Father</th>
+                </tr>
+              </thead>
+
+              <tbody>
+                {people.map(person => {
+                  const mother = people.find(
+                    per => per.name === person.motherName,
+                  );
+                  const father = people.find(
+                    per => per.name === person.fatherName,
+                  );
```

Comment:
It would be better to create separate  function for this.

Reply:
Good point, fix it